packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -203,3 +206,80 @@ export function convertToSummaryTree(
         return builder.getSummaryTree();
     }
 }
+
+/**
+ * Converts ISnapshotTree to ISummaryTree format and tracks stats.
+ * @param snapshot - snapshot in ISnapshotTree format
+ */
+export function convertSnapshotTreeToSummaryTree(
+    snapshot: ISnapshotTree,
+): ISummaryTreeWithStats {
+    const blobMapInitial = new Map(Object.entries(snapshot.blobs));
+    const blobMapFinal = new Map<string, string>();
+    for (const [key, value] of blobMapInitial.entries()) {
+        if (blobMapInitial.has(value)) {
+            blobMapFinal[key] = blobMapInitial.get(value);
+        }
+    }
```

Comment:
I'm a little confused what this block of code is doing?
What is the difference between `blobMapInitial` and `blobMapFinal`?

Reply:
I am not sure actually. I moved and updated this from here - https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/datastore/src/utils.ts.
@jatgarg should be able to answer this question.

Reply:
I think this part is redundant and should be removed, but we can wait and see what @jatgarg says.

Reply:
This api is used while rehydrating the container from detached container snapshot. In there the blobs in the detached snapshot needs to be converted in required format(as it contains some extra data) so that the it can be used properly while rehydration. So blob map initial is the actual blobs present in detached snapshot and final is the one which is actually converted to  ISummary tree. 

---
In reply to: [518980381](https://github.com/microsoft/FluidFramework/pull/4230#discussion_r518980381) [](ancestors = 518980381)

Reply:
Okay, but specifically in these 7 lines of code, `blobMapInitial` simply contains entries one-to-one with `snapshot.blobs`, so basically converted from plain JS object to Map.

Then loop through those entries in `blobMapInitial` and add them to `blobMapFinal` if they exist in `blobMapInitial` (redundant?). Seems like `blobMapInitial` and `blobMapFinal` are the same.

Reply:
No, there is a if condition inside for loop.
        if (blobMapInitial.has(value)) {
            blobMapFinal[key] = blobMapInitial.get(value);
        }

Reply:
@jatgarg Do you have an example of where they would differ? I can use that to test the API and verify this.

Reply:
Actually I guess this is unrelated to your change as you are just moving stuff. I will have a PR myself to simplify this just don't want to break anything. So you can either just merge your PR or move this api back to previous file.(maybe just merge the PR).

Reply:
Okay. I will move it for now. Can you create an issue to track this?

Reply:
https://github.com/microsoft/FluidFramework/issues/4278

Reply:
Oh I see the difference now, it is indexing on `value` instead of `key`. Not sure the intention here, but it does seem suspicious to me as it only goes 1 level deep. a -> b, b -> c, c -> d would give a -> c, b -> d.
Sorry I misread it the first time, thought it was indexing on `key`.